python/ambassador/envoy/v2/v2ready.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2021 Datawire. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+from typing import TYPE_CHECKING
+
+from .v2listener import V2Listener
+
+if TYPE_CHECKING:
+    from . import V2Config # pragma: no cover
+
+
+class V2Ready(dict):
+
+    @classmethod
+    def generate(cls, config: 'V2Config') -> None:
+        # Inject the ready listener to the list of listeners if enabled
+        rport = config.ir.aconf.module_lookup('ambassador', 'ready_port', -1)
+        if rport <= 0:
+            return
+
+        ready_listener = {
+            'name': 'ambassador-listener-ready-%s' % rport,
+            'address': {
+                'socket_address': {
+                    'address': '0.0.0.0', # Todo: Change this to 127.0.0.1 or make it a parameter
+                    'port_value': rport,
+                    'protocol': 'TCP'
+                }
+            },
+            'filter_chains': [
+                {
+                    'filters': [
+                        {
+                            'name': 'envoy.filters.network.http_connection_manager',
+                            'typed_config': {
+                                '@type': 'type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager',
+                                "access_log": [
+                                    {
+                                        "name": "envoy.access_loggers.file",
+                                        "typed_config": {
+                                            "@type": "type.googleapis.com/envoy.config.accesslog.v2.FileAccessLog",
+                                            "format": "ACCESS [%START_TIME%] \"%REQ(:METHOD)% %REQ(X-ENVOY-ORIGINAL-PATH?:PATH)% %PROTOCOL%\" %RESPONSE_CODE% %RESPONSE_FLAGS% %BYTES_RECEIVED% %BYTES_SENT% %DURATION% %RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)% \"%REQ(X-FORWARDED-FOR)%\" \"%REQ(USER-AGENT)%\" \"%REQ(X-REQUEST-ID)%\" \"%REQ(:AUTHORITY)%\" \"%UPSTREAM_HOST%\"\n",
```

Comment:
I'm kind of torn about logging for this, honestly. 🙂  If it's present, though, I think it should honor the existing logging configuration stuff, e.g. JSON logging if that's enabled, custom formats, etc...

Reply:
Thanks for the review.

You are totally right, I need to honor the logging config.

I want to add one more config info as well, regarding the ip binding.
Let me give more details on the 3 changes I'd like to make:

1/ Make the logging honor the existing logging config (default)
2/ Add an option to disable logging completely for this listener.
3/ Add an option for which ip is binded by the listener. By default I'll make it 127.0.0.1.

For this config, I'd like to make the config an object in the module config with "ready_listener" as the key and the following keys for this object:

```
ready_listener:
  port: -1 #-1 disables this listener
  nolog: false # disable logging entirely if "true"
  ip: 127.0.0.1
```

Reason for 3/: 
This listener should not need to be exposed, so 127.0.0.1 makes sense.
But as it is not used yet by diagd and the go code, exposing this listener to the k8s readiness means it must bind the external ip of the pod. That's why the option is present o change it (to 0.0.0.0 for instance) as a first step.

I'll try to update the review with those changes, unless you have some feedback.

Thanks.

Reply:
As in the long term, I see this enabled and used all the time, I'm wondering if an object makes sense.
Maybe just:
```
 ready_port:
 ready_log: true # default true
```
could be enough?